### PR TITLE
feat(cli/templates) add `vanilla-ts`

### PR DESCRIPTION
**Pull request by 0xle0ne**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/create-tauri-app/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### Other information

Related to this https://github.com/tauri-apps/create-tauri-app/issues/155

### CODE REVIEW

File: packages/cli/fragments/fragment-vanilla-ts/index.html [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    <div class="container">
+      <h1>Welcome to Tauri!</h1>
+
+      <div class="row">
+        <a href="https://tauri.app" target="_blank">
+          <img
+            src="/src/assets/tauri.svg"
+            class="logo tauri"
+            alt="Tauri logo"
+          />
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-vanilla-ts/index.html:33]: move the tauri logo to the middle

File: packages/cli/fragments/fragment-vanilla-ts/_[pnpm-yarn-npm]_package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-vanilla-ts/_[pnpm-yarn-npm]_package.json]: rename this file to just `package.json`

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: packages/cli/fragments/fragment-vanilla-ts/package.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  "type": "module",
+  "scripts": {
+    "dev": "vite",
+    "build": "tsc && vite build",
+    "preview": "vite preview",
+    "tauri": "tauri"
+  },
+  "dependencies": {
+    "@tauri-apps/api": "^1.0.2"
+  },
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-vanilla-ts/package.json]: remove this, we should use global tauri

File: packages/cli/fragments/fragment-vanilla-ts/src/main.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { invoke } from "@tauri-apps/api";
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: use global tauri instead. `window.__TAURI__`

```suggestion
const { invoke } = window.__TAURI__;
```

File: packages/cli/fragments/fragment-vanilla-ts/src/main.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+declare global {
+  interface Window {
+    greet: () => void;
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: add type definitions for `window.__TAURI__`

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: packages/cli/fragments/fragment-vanilla-ts/src/main.ts (hunk 5 above)
Comment (0xle0ne) [reply packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: You want me to redeclare all functions types for `window.__TAURI__` ?

Comment (amrbashir) [reply packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: no, just nivoke

Comment (0xle0ne) [reply packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: What about using `@tauri-apps/api` as `dev` dependency and do something like this:
```ts
import type * as TauriApiTypes from "@tauri-apps/api";

declare global {
  interface Window {
    greet: () => void;
    __TAURI__: typeof TauriApiTypes
  }
}

const { invoke } = window.__TAURI__;

let greetInputEl: HTMLInputElement | null;
let greetMsgEl: HTMLElement | null;

window.addEventListener("DOMContentLoaded", () => {
  greetInputEl = document.querySelector("#greet-input");
  greetMsgEl = document.querySelector("#greet-msg");
});

async function greet() {
  if (greetMsgEl && greetInputEl) {
    greetMsgEl.textContent = await invoke("greet", {
      name: greetInputEl.value,
    });
  }
}

window.greet = greet;
```

That way if you change something in the package `@tauri-apps/api` it's will update there too. And it wont be added in the final bundle.

Comment (amrbashir) [reply packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: I like that, but could you extract that in a shim file outside `main.ts` ?

Comment (amrbashir) [reply packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: something like this https://github.com/vitejs/vite/blob/main/packages/create-vite/template-vue-ts/src/vite-env.d.ts

Comment (0xle0ne) [reply packages/cli/fragments/fragment-vanilla-ts/src/main.ts]: Yeah sure sounds good to me i have updated.

File: .changes/vanilla-ts.md [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+"create-tauri-app": "patch"
+"create-tauri-app-js": "patch"
```
Comment (amrbashir) [inline .changes/vanilla-ts.md]:
```suggestion
"create-tauri-app": "minor"
"create-tauri-app-js": "minor"
```

**Review by amrbashir — APPROVED** [review]
Thank you

**Pull request merged**